src/server/controllers/addNewUser.js

```diff
@@ -1,11 +1,30 @@
 const { createUser } = require('./../../database/db_create');
+const sendEmail = require('./../../helpers/emailVerification');
 
 module.exports = (req, res) => {
+  // create random 16 character token for email verification
+  const chars = '0123456789abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ';
+  let token = '';
+  for (let i = 16; i > 0; --i) {
+    token += chars[Math.round(Math.random() * (chars.length - 1))];
+  }
+
+  req.body.token = token;
+
   createUser(req.body, (error, response) => {
     if (error) {
       // will be refactored, so that err.message is displayed on screen
       res.status(400).send({ error: error.message });
     }
     res.redirect('/');
   });
+
+  sendEmail(req.body.username, req.body.email, token, (error, info) => {
+    if(error) {
+      res.status(400).send({ error: error.message });
```

Comment:
Okay, this is better, we are close!

Right now you are still sending the email and making the database call at the same time. If they both fail you are going to try and send 2 errors, I is actually impossible to know in this case which will be fired (the keyword here is `race condition`).

You have two options.

One: Run these in parallel but handle the errors correctly (more complicated, unnecessary ).
```js
// https://www.npmjs.com/package/async
async.parallel([
  cb => createUser(req.body, cb)
  cb => sendEmail(req.body.username, req.body.email, token, cb)
], (err, results) => {
  if (err is from db) {send an error}
  else if (err is from ses) { send some other error }
  else { success! }
})
```
Two: run these in series
```js
createUser(req.body, (dbError, dbResponse) => {
  if (dbError) {
    res.status(500).send ..
   return 
  }
  sendEmail(req.body.username, req.body.email, token, (SESError, SESResponse) => {
    if (SESError) {
      res.status(500).send ..
      return
    }
    res.send('success')
  })
})
 ```

You could use async.waterfall but unnecessary here.

In the second example, you are only sending an email the the db call succeeds, which I expect is the correct behaviour.